test/odbc/psqlodbc/psqlodbc_tests_common.h

```diff
@@ -0,0 +1,696 @@
+  
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affectedRows);
+  EXPECT_EQ(rcode, SQL_SUCCESS);
+  EXPECT_EQ(affectedRows, insertedValues.size());
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(tableName, {"*"}, vector<string> {orderByColumnName}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < insertedValues.size(); i++) {
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    EXPECT_EQ(rcode, SQL_SUCCESS);
+    EXPECT_EQ(pk_len, 4);
+    EXPECT_EQ(pk, i);
+    if (insertedValues[i] != "NULL") {
+      EXPECT_EQ(data_len, expectedLen[i]);
+      EXPECT_EQ(data, expectedInsertedValues[i]);
+    }
+    else {
+      EXPECT_EQ(data_len, SQL_NULL_DATA);
+    }
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  EXPECT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", tableName));
+}
+
+template <typename T>
+void testComparisonFunctionsNumeric(string tableName, vector<pair<string, string>> tableColumns, string orderByColumnName, int type, vector<T> colResults, int bufferLen, vector<string> insertedValues, vector<string> operationsQuery, vector<T> expectedResults, vector<long> expectedLen) {
+  
+  RETCODE rcode;
+  SQLLEN affected_rows;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  const int NUM_OF_DATA = insertedValues.size();
+  const int NUM_OF_OPERATIONS = operationsQuery.size();
+  SQLLEN col_len[NUM_OF_OPERATIONS];
+
+ vector<tuple<int, int, SQLPOINTER, int, SQLLEN *>> bind_columns = {};
+
+  // initialization for bind_columns
+  for (int i = 0; i < NUM_OF_OPERATIONS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN *> tuple_to_insert(i + 1, type, (SQLPOINTER)&colResults[i], bufferLen, &col_len[i]);
+    bind_columns.push_back(tuple_to_insert);
+  }
+
+  string insert_string{};
+  string comma{};
+
+  // insert_string initialization
+  for (int i = 0; i < NUM_OF_DATA; i++) {
+    insert_string += comma + "(" + std::to_string(i) + "," + insertedValues[i] + ")";
+    comma = ",";
+  }
+
+  // Create table
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(tableName, tableColumns));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(tableName, insert_string));
+
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  EXPECT_EQ(rcode, SQL_SUCCESS);
+  EXPECT_EQ(affected_rows, NUM_OF_DATA);
+
+  // Make sure inserted values are correct and operations
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  odbcHandler.CloseStmt();
```

Comment:
It looks like the pattern of creating a table, inserting data and verifying number of rows inserted is used in most of the helper functions with the same code repeated.

It is probably a good idea create a separate function that does that. 
The function would take a reference to OdbcHandler and the data needed for table creation and data insertion

This function could potentially handle both numeric and char data.

Another advantage of this (I think) is that it would be easier to handle scenarios where table is created and data inserted using bbf connection and the rest of the test is using postgres connection and vice versa.
 
Given this thought, the call to the new create/insert function could happen in the test itself so that different tests can choose scenario and the helper functions (e.g. testComparisonFunctionsChar) would do all the current steps minus table creation/data insertion. They would assume the table and relevant data is there.
That is the hel

Reply:
Agreed. Done, I broke the code into chunks & have separate functions that insert values in tables, & verify the inserted values in the tables.

Also made it so that the tests themselves perform the necessary steps required. For example:
```
TEST_F(Test_PrimaryKeys) {
  createTable(ServerType, tableName);
  insertAndVerifyValues(ServerType, tableName, inserted, expected);
  testPrimaryKeys(ServerType, tableName)l
  dropTable(ServerType, tableName)
}

```